packages/lesswrong/server/resolvers/tagResolvers.ts

```diff
@@ -176,10 +158,134 @@ addGraphQLSchema(`
 
 addGraphQLResolvers({
   Mutation: {
-    async recordSubforumView(root: void, {userId, tagId}: {userId: string, tagId: string}, context: ResolverContext) {
-      await recordSubforumView(userId, tagId);
-      return "success";
-    }
+    async mergeTags(
+      root: void,
+      {
+        sourceTagId,
+        targetTagId,
+        transferSubtags,
+        redirectSource,
+      }: { sourceTagId: string; targetTagId: string; transferSubtags: boolean; redirectSource: boolean },
+      context: ResolverContext
+    ) {
+      const { currentUser } = context;
+
+      if (!userIsAdminOrMod(currentUser)) {
+        throw new Error(`Must be an admin/mod to merge ${taggingNamePluralSetting.get()}`);
+      }
+      if (!sourceTagId || !targetTagId) {
+        throw new Error("sourceTagId and targetTagId required");
+      }
+
+      const sourceTag = await Tags.findOne({ _id: sourceTagId });
+      const targetTag = await Tags.findOne({ _id: targetTagId });
```

Comment:
Minor, but we could `Promise.all` these to make it a bit more parallel. The same goes for fetching the tag rels below.

Reply:
Hm, given that there is no need for this to be that fast I think I'll keep it as is to make it somewhat easier to debug